### ISSUE TITLE
[Suggestion] Winning Wave

### ISSUE THREAD

**Issue by fireheart3911** [garbagemule/MobArena#735]

**Comment by fireheart3911** [comment]
One sec, github blocked the content

**Comment by fireheart3911** [comment]
Note: The Placeholder Expansion suggests that something like that already exists, i'm just unable to find it on the wiki.

**Comment by Nesseley** [comment]
https://github.com/garbagemule/MobArena/wiki/Setting-up-the-config-file#settings 
`final-wave: x`

**Comment by garbagemule** [comment]
It seems like most of this is possible with the final wave setting and the various clear wave flags, as well as command rewards. Whatever isn't possible currently might be when #678 is implemented. Perhaps it's worth first exploring the more generic functionality of the plugin before we dive into hyper-specific functionality for "winning", which isn't really a concept in MobArena in the first place (but very likely will be in the Sessions Rework).

**Comment by fireheart3911** [comment]
> It seems like most of this is possible with the final wave setting and the various clear wave flags
What i meant with that was a new option to only clear all mobs in the final wave and nowhere else

**Comment by garbagemule** [comment]
Right, if you want to let waves progress at a specific rate (`clear-wave-before-next: false`), but halt the progression on a certain wave to sort of mix with the clear-based configuration (`clear-wave-before-next: true`), there are a couple of ways to sort of emulate that, but we don't have a very flexible solution that would accommodate this in the general case.

The idea of making the final wave the only _to clear_ has its merits, but it's really just a specialization or "inversion" of the idea suggested in #259. I think the wave system is generally just too inflexible, and that's exactly what the Sessions Rework is meant to change. I'll have to give it a bit more thought, because introducing _yet another_ per-arena setting for _yet another_ very specific use case really is the bane of the Sessions Rework. On the other hand, it seems fairly straightforward, so it might be worth another look.

In the meantime, you might be able to emulate the idea if you don't have any boss waves, since boss waves do have a clear before flag?

**Comment by fireheart3911** [comment]
Update:
 
- ~~Customizable Win Title~~ Possible with Reward CMDs
- ~~Fireworks~~ Possible with Reward CMDs 
-~~ Completion Percentage in scoreboard~~ Possible with Reward CMDs
- Delay before teleport (Semi-solved, needs a very provisional workaround with delayed functions)
- ~~Option if you "just" have to survive or to clear all mobs~~ Possible with Reward CMDs
- Leaderboard/Placeholders for completion time - This could be implemented in MobArenaStats
